index.js

```diff
@@ -6,8 +6,8 @@ var util = require('util');
  * Regexps
  */
 
-var DATE_TYPE_RE = /^\d{4}\-\d{2}\-\d{2}$/;
-var DATETIME_TYPE_RE = /^\d{4}\-\d{2}\-\d{2} \d{2}:\d{2}:\d{2}$/;
+var DATE_TYPE_RE = /^(?:[1-9]\d{3}-(?:(?:0[1-9]|1[0-2])-(?:0[1-9]|1\d|2[0-8])|(?:0[13-9]|1[0-2])-(?:29|30)|(?:0[13578]|1[02])-31)|(?:[1-9]\d(?:0[48]|[2468][048]|[13579][26])|(?:[2468][048]|[13579][26])00)-02-29)$/;
```

Comment:
Original regexp is [here](https://stackoverflow.com/questions/26355985/regular-expression-for-ddmmyyyy-date-including-validation-for-leap-year).

I changed it to match format - `YYYY-MM-DD hh:mm:ss`. A diagram can be found [here](https://www.debuggex.com/r/CqiuCg-1JZBfXr8o).

I test it with `benchmark.js` bundled with this lib:

```
  globalParameter: verify {"date":"date"}                                          x 4,966,946 ops/sec ±1.31% (85 runs sampled)
  localParameter: verify {"date":"date"}                                           x 4,485,268 ops/sec ±0.49% (93 runs sampled)
  globalParameter: verify {"date":{"type":"date"}}                                 x 4,992,992 ops/sec ±0.66% (94 runs sampled)
  localParameter: verify {"date":{"type":"date"}}                                  x 4,350,743 ops/sec ±0.53% (94 runs sampled)
  globalParameter: verify {"time":"datetime"}                                      x 4,957,236 ops/sec ±0.66% (95 runs sampled)
  localParameter: verify {"time":"datetime"}                                       x 4,289,336 ops/sec ±0.75% (92 runs sampled)
  globalParameter: verify {"time":{"type":"datetime"}}                             x 4,716,744 ops/sec ±1.28% (94 runs sampled)
  localParameter: verify {"time":{"type":"datetime"}}                              x 4,201,725 ops/sec ±1.02% (93 runs sampled)
```